#! /usr/bin/env python

PACKAGE='homework4'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type       Lvl    Description                                   Default     Min     Max
gen.add("max_match_dist",   double_t,  0,     "Maximum distance to associate measurement",  3.0,        0.1,    10.0)
gen.add("min_size_z",       double_t,  0,     "Minimum z dimension to track",               0.3,        0.1,    20.0)
gen.add("y_window",         double_t,  0,     "Lateral distance of tracking window",        10.0,       0.1,    200.0)
gen.add("min_age",          double_t,  0,     "Minimum age for output",                     1.0,        0.0,    5.0)
gen.add("q_pos",            double_t,  0,     "Kalman state standard deviation",            2.0,        1e-6,   100.0)
gen.add("q_vel",            double_t,  0,     "Kalman state standard deviation",            5.0,        1e-6,   100.0)
gen.add("r_pos",            double_t,  0,     "Kalman measurement standard deviation",      0.5,        1e-6,   100.0)

exit(gen.generate(PACKAGE, PACKAGE, "Homework4"))
